### PR TITLE
[FIXED] MQTT sess.clear: bring deleteConsumers back under the sess lock

### PR DESCRIPTION
**Pull request by levb**
A customer mentioned in a Zendesk ticket, "_It is also not possible to clean sessions._" (in their cluster configuration).

One of the things that changed in 2.10 session cleaning was moving the JS calls to delete Subscription's consumers outside the session lock, see 2.9.x implementation [here](https://github.com/nats-io/nats-server/blob/release/v2.9.23/server/mqtt.go#L2338).

Not 100% sure why it would matter with a cluster, but bringing the JS calls back under the lock, to be safe.

### CODE REVIEW

**Review by kozlovic — APPROVED** [review]
LGTM

File: server/mqtt.go [hunk 1]
```diff
@@ -2676,20 +2676,23 @@ func (sess *mqttSession) clear() error {
 	sess.pubRelConsumer = nil
 	sess.seq = 0
 	sess.tmaxack = 0
-	sess.mu.Unlock()
 
 	for _, dur := range durs {
 		if _, err := sess.jsa.deleteConsumer(mqttStreamName, dur); isErrorOtherThan(err, JSConsumerNotFoundErr) {
+			sess.mu.Unlock()
```
Comment (derekcollison) [inline server/mqtt.go:2682]: This means we are holding the lock across and request/reply boundary, which will be hopping between servers and routes. So do not think we should do that, unless I am mistaken on deleteConsumer. But that is also why we do not hold below for deleteMsg.

**Review by derekcollison — CHANGES_REQUESTED** [review]
I don't think we should do this.

What was it seeming to solve?

File: server/mqtt.go (hunk 1 above)
Comment (levb) [reply server/mqtt.go:2682]: @derekcollison Understood, that's why I made the change to begin with, in https://github.com/nats-io/nats-server/pull/4349/files#diff-ef6be3a4d0880e5771af5ea89b210064b3f988fdde90076a188b3b20360d8a13R2664 but it was not really related to QoS2; creeped in.

This PR would revert to how it was in v2.9

Anyway, we can hold off on this PR, see if https://github.com/nats-io/nats-server/pull/4734 addresses session `clear()` concerns.

**Pull request closed without merging**